### PR TITLE
Option to allow expired certs

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

**Comment by turt2live** [comment]
Link to docs that says this works?

**Comment by Half-Shot** [comment]
https://github.com/matrix-org/node-irc/blob/master/lib/irc.js#L867

**Comment by turt2live** [comment]
Generally it looks safe, but I think it needs to be added to the schema and example config.

**Comment by Half-Shot** [comment]
That it does. Halfy does so.

File: config.sample.yaml [hunk 1]
```diff
@@ -58,6 +58,9 @@ ircService:
       ssl: true
       # Whether or not IRC server is using a self-signed cert or not providing CA Chain
       sslselfsign: false
+      # Whether to allow expired certs when connecting to the IRC server.
+      # Usually this should be off. Default: false.
```
Comment (turt2live) [inline config.sample.yaml]: a comment about how this interacts with/against `sslselfsign` would be nice.

Comment (Half-Shot) [reply config.sample.yaml]: I think I'm failing to see how they intersect. This is about expired certs, not ones that might be invalid due to self signing?

Comment (turt2live) [reply config.sample.yaml]: because I'm an idiot and read the wrong words. Ignore me on this comment - LGTM overall.

Comment (Half-Shot) [reply config.sample.yaml]: :) *pats travis*

**Pull request merged**